src/compiler/scala/tools/nsc/typechecker/Analyzer.scala

```diff
@@ -80,8 +80,17 @@ trait Analyzer extends AnyRef
 
       def apply(unit: CompilationUnit): Unit = {
         openPackageObjectsTraverser(unit.body)
-        deferredOpen.foreach(openPackageModule(_))
-        deferredOpen.clear()
+      }
+
+      override def run(): Unit = try {
+        super.run()
+      } finally {
+        val toOpen = deferredOpen.toVector
+        for (sym <- toOpen) {
+          if (deferredOpen.remove(sym))
+            // CAREFUL -- this can mutate deferredOpen, hence the defensive code here!
```

Comment:
can it add more symbols to `deferredOpen` that also need to be opened?

Reply:
AFAICT it only removes them